cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,179 @@
+Course Apps API
+_______________
+
+Status
+======
+Proposal
+
+Context
+=======
+
+The new _Course Authoring MFE_includes a new UX called "Pages and Resources"
+for configuring course "apps" such as progress, wiki, teams, discussions,
+etc. all from one place. Currently, these apps are intermingled with the
+core platform and are inseparable from it, however, future apps could be
+implemented as plugins.
+
+This ADR proposes a way to configure these apps from the backend that MFEs
+can use and potentially enable/disable these apps using an API.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We can introduce a new type of Open edX plugin, a course app. These are
+course-level apps that add new functionality to the course for students.
+Each course app has associated metadata and a standard API to determine if
+the app is available, enabled, etc.
+
+This metadata should be made available via a configuration class similar to
+that used for Django apps, or course tabs etc. We can develop a new course
+apps API that can be driven by the metadata provided by these installed
+plugins.
+
+In the context of these apps, we need to distinguish between *installed*,
+*available* and *enabled*. The standard plugin API will automatically
+discover all installed plugins. However if a plugin is installed for a
+select few courses, there should be a way to only enable it for those
+courses. To do this we use have a mechanism to determine availability. An
+app should provide a method to check if it should be made available for
```

Comment:
=> `An app should provide a method to check if it should be made available for ...`
Nit: Maybe easier if we 

`Each app should expose a helper method to check its availability for a course.`